lib/matplotlib/lines.py

```diff
@@ -788,10 +792,13 @@ def draw(self, renderer):
 
                 alt_marker_path = marker.get_alt_path()
                 if alt_marker_path:
-                    if rgbaFaceAlt is not None:
-                        gc.set_alpha(rgbaFaceAlt[3])
                     alt_marker_trans = marker.get_alt_transform()
                     alt_marker_trans = alt_marker_trans.scale(w)
+                    if (is_string_like(mec) and mec == 'auto' and
+                            rgbaFaceAlt is not None):
+                        gc.set_alpha(rgbaFaceAlt[3])
+                    else:
+                        gc.set_alpha(self.get_alpha())
```

Comment:
Should this go in a separate function for maintainability?  Virtually the same code as above, just with ``rgbaFace`` replaced with ``rgbaFaceAlt``.